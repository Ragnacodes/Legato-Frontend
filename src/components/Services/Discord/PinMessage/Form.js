import React, { useState, useLayoutEffect } from 'react';
import { MenuItem, Button, TextField, IconButton } from '@material-ui/core';
import { Refresh, Search } from '@material-ui/icons';
import { connect } from 'react-redux';
import { startGetConnections } from '../../../../actions/connections';
import * as actions from '../../../../actions/discord';
import ServiceForm from '../../../PopoverForm';
import MessageList from '../MessageList';
import Message from '../Message';
export function Form({
  id,
  data,
  connections,
  channels,
  messages,
  editElement,
  setAnchorEl,
  getChannels,
  getConnections,
  getMessages,
}) {
  const [info, setInfo] = useState({
    connection: data.connection || '',
    channelId: data.channelId || '',
    messageId: data.messageId || '',
  });

  const [guildId, setGuildId] = useState('');
  const [channelLoading, setChannelLoading] = useState(true);
  const [message, setMessage] = useState({
    component: '',
    loading: true,
    anchor: null,
    error: false,
  });

  useLayoutEffect(() => {
    if (guildId)
      getChannels(guildId).then(() => {
        setChannelLoading(false);
      });
  }, [guildId, getChannels]);

  useLayoutEffect(() => {
    if (info.channelId)
      getMessages(info.channelId).then(() => {
        setMessage((prev) => ({ ...prev, loading: false }));
      });
  }, [info.channelId, getMessages]);

  useLayoutEffect(() => {
    getConnections();
  }, [getConnections]);

  useLayoutEffect(() => {
    if (connections.length) {
      setGuildId(connections[0].data.guildId);
    }
  }, [connections]);

  const handleChange = (e) => {
    setInfo((prev) => ({
      ...prev,
      [e.target.name]: e.target.value,
    }));
  };

  const handleCancel = () => {
    setAnchorEl(null);
    setInfo({
      connection: data.connection || '',
      channelId: data.channelId || '',
      messageId: data.messageId || '',
    });
  };

  const handleSave = () => {
    const updates = {
      name: info.name,
      data: { ...data, ...info },
    };
    editElement(id, updates);
    setAnchorEl(null);
  };

  const openMessageList = (e) => {
    setMessage((prev) => ({ ...prev, anchor: e.currentTarget }));
  };

  let disabledSave = !info['messageId'] || !info['channelId'];

  const findMessage = (id) => {
    const msg = messages.find((message) => message.id === id);
    setMessage((prev) => ({ ...prev, component: msg }));
    if (!msg) setMessage((prev) => ({ ...prev, error: true }));
    else setMessage((prev) => ({ ...prev, error: false }));
  };

  const saveMessageId = (messageId) => {
    setInfo((prev) => ({ ...prev, messageId: messageId }));
    findMessage(messageId);
  };

  return (
    <ServiceForm
      className="discord-forms"
      title="Pin a message"
      disabledSave={disabledSave}
      handleSave={handleSave}
      handleCancel={handleCancel}
    >
      <MessageList
        messages={messages}
        anchor={message.anchor}
        setAnchor={(change) =>
          setMessage((prev) => ({ ...prev, anchor: change }))
        }
        value={info['messageId']}
        handleSave={saveMessageId}
      />

      <div className="channel-field">
        {channelLoading ? (
          <TextField
            disabled
            className="text-field"
            size="small"
            label="Channel"
            value="Loading..."
            variant="outlined"
          />
        ) : (
          <TextField
            name="channelId"
            className="text-field"
            size="small"
            select
            label="Channel"
            value={info['channelId']}
            onChange={handleChange}
            variant="outlined"
          >
            {channels.map((c) => (
              <MenuItem key={c.id} value={c.id}>
                {c.name}
              </MenuItem>
            ))}
          </TextField>
        )}

        <IconButton
          name="refreshChannels"
          size="small"
          className="add-icon"
          onClick={() => {
            setChannelLoading(true);
            getChannels().then(() => {
              setChannelLoading(false);
            });
          }}
          disabled={channelLoading}
        >
          <Refresh />
        </IconButton>
      </div>

      {info['channelId'] && !message.loading && (
        <div>
          <div className="channel-field">
            <TextField
              name="messageId"
              className="text-field"
              size="small"
              label="Message Id"
              value={info['messageId']}
              onChange={handleChange}
              variant="outlined"
              helperText={message.error && 'Unable to find the message.'}
              error={message.error}
            />
            <IconButton
              name="refreshChannels"
              size="small"
              className="add-icon"
              onClick={() => findMessage(info['messageId'])}
              disabled={message.loading}
            >
              <Search />
            </IconButton>
          </div>
          <Button size="small" color="primary" onClick={openMessageList}>
            or choose from the list.
          </Button>
        </div>
      )}

      {info['messageId'] && !!message.component && (
        <Message message={message.component} />
      )}
    </ServiceForm>
  );
}
const mapStateToProps = (state) => ({
  connections: state.connections,
  channels: state.discord.channels,
  messages: state.discord.messages,
});

const mapDispatchToProps = (dispatch) => ({
  getConnections: () => dispatch(startGetConnections()),
  getChannels: (guildId) => dispatch(actions.startGetChannels(guildId)),
  getMessages: (channelId) => dispatch(actions.startGetMessages(channelId)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Form);
