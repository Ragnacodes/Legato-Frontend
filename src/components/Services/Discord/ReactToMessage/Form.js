import React, { useState, useLayoutEffect } from 'react';
import { MenuItem, Button, TextField, IconButton } from '@material-ui/core';
import { Refresh } from '@material-ui/icons';
import { connect } from 'react-redux';
import { startGetConnections } from '../../../../actions/connections';
import * as actions from '../../../../actions/discord';
import ServiceForm from '../../../PopoverForm';
import MessageList from '../MessageList';
import Message from '../Message';
export function Form({
  id,
  data,
  connections,
  channels,
  messages,
  editElement,
  setAnchorEl,
  getChannels,
  getConnections,
  getMessages,
}) {
  const [info, setInfo] = useState({
    connection: data.connection || '',
    channelId: data.channelId || '',
    messageId: data.messageId || '',
    react: data.react || '',
  });

  const [guildId, setGuildId] = useState('');

  useLayoutEffect(() => {
    if (guildId) getChannels(guildId);
  }, [guildId, getChannels]);

  useLayoutEffect(() => {
    if (info.channelId) getMessages(info.channelId);
  }, [info.channelId, getMessages]);

  useLayoutEffect(() => {
    getConnections();
  }, [getConnections]);

  useLayoutEffect(() => {
    if (connections.length) {
      setGuildId(connections[0].data.guildId);
    }
  }, [connections]);

  const handleChange = (e) => {
    setInfo((prev) => ({
      ...prev,
      [e.target.name]: e.target.value,
    }));
  };

  const handleCancel = () => {
    setAnchorEl(null);
    setInfo({
      connection: data.connection || '',
      channelId: data.channelId || '',
      messageId: data.messageId || '',
      react: data.react || '',
    });
  };

  const handleSave = () => {
    const updates = {
      name: info.name,
      data: { ...data, ...info },
    };
    editElement(id, updates);
    setAnchorEl(null);
  };
  const [messageListAnchor, setMessageListAnchor] = useState(null);

  const openMessageList = (e) => {
    setMessageListAnchor(e.currentTarget);
  };

  let disabledSave = !info['messageId'] || !info['channelId'] || !info['react'];

  const FindMessage = (id) => {
    const msg = messages.find((message) => message.id === id);
    if (msg) return msg;
    return {};
  };

  const saveMessageId = (messageId) => {
    setInfo((prev) => ({ ...prev, messageId: messageId }));
  };

  return (
    <ServiceForm
      className="discord-forms"
      title="React to a message"
      disabledSave={disabledSave}
      handleSave={handleSave}
      handleCancel={handleCancel}
    >
      <MessageList
        messages={messages}
        anchor={messageListAnchor}
        setAnchor={setMessageListAnchor}
        value={info['messageId']}
        handleSave={saveMessageId}
      />

      <div className="channel-field">
        <TextField
          name="channelId"
          className="text-field"
          size="small"
          select
          label="Channel"
          value={info['channelId']}
          onChange={handleChange}
          variant="outlined"
        >
          {channels.map((c) => (
            <MenuItem key={c.id} value={c.id}>
              {c.name}
            </MenuItem>
          ))}
        </TextField>

        <IconButton
          name="refreshChannels"
          size="small"
          className="add-icon"
          onClick={getChannels}
        >
          <Refresh />
        </IconButton>
      </div>

      <TextField
        name="react"
        className="text-field"
        size="small"
        label="Reaction"
        value={info['react']}
        helperText="Choose an emoji."
        onChange={handleChange}
        variant="outlined"
      />

      <Button
        onClick={openMessageList}
        disabled={false}
        fullWidth
        variant="contained"
        color="primary"
      >
        Choose a message
      </Button>
      {info['messageId'] && (
        <Message message={FindMessage(info['messageId'])} />
      )}
    </ServiceForm>
  );
}
const mapStateToProps = (state) => ({
  connections: state.connections,
  channels: state.discord.channels,
  messages: state.discord.messages,
});

const mapDispatchToProps = (dispatch) => ({
  getConnections: () => dispatch(startGetConnections()),
  getChannels: (guildId) => dispatch(actions.startGetChannels(guildId)),
  getMessages: (channelId) => dispatch(actions.startGetMessages(channelId)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Form);
