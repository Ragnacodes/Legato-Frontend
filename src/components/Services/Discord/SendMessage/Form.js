import React, { useState, useEffect } from 'react';
import { MenuItem, IconButton, TextField } from '@material-ui/core';
import { Refresh } from '@material-ui/icons';
import { connect } from 'react-redux';
import { startGetConnections } from '../../../../actions/connections';
import * as actions from '../../../../actions/discord';
import ServiceForm from '../../../PopoverForm';
export function Form({
  id,
  data,
  connections,
  channels,
  editElement,
  setAnchorEl,
  getChannels,
  getConnection,
}) {
  const [info, setInfo] = useState({
    connection: data.connection || '',
    channelId: data.channelId || '',
    content: data.content || '',
  });

  const [guildId, setGuildId] = useState('');

  // useLayoutEffect(() => {
  //   if (guildId) getChannels(guildId);
  // }, [guildId, getChannels]);

  // useLayoutEffect(() => {
  //   getConnection();
  // }, [getConnection]);

  // useLayoutEffect(() => {
  //   if (connections) {
  //     setGuildId(connections[0].data.guildId);
  //   }
  // }, [connections]);

  const handleChange = (e) => {
    setInfo((prev) => ({
      ...prev,
      [e.target.name]: e.target.value,
    }));
  };

  const handleCancel = () => {
    setAnchorEl(null);
    setInfo({
      connection: data.connection || '',
      channelId: data.channelId || '',
      content: data.content || '',
    });
  };

  const handleSave = () => {
    const updates = {
      name: info.name,
      data: { ...data, ...info },
    };
    editElement(id, updates);
    setAnchorEl(null);
  };

  let disabledSave = !info['channelId'] || !info['content'];

  return (
    <ServiceForm
      className="discord-forms"
      title="Post a message"
      disabledSave={disabledSave}
      handleSave={handleSave}
      handleCancel={handleCancel}
    >
      <div className="channel-field">
        <TextField
          name="channelId"
          className="text-field"
          size="small"
          select
          label="Channel"
          value={info['channelId']}
          onChange={handleChange}
          variant="outlined"
        >
          {channels.map((c) => (
            <MenuItem key={c.id} value={c.id}>
              {c.name}
            </MenuItem>
          ))}
        </TextField>

        <IconButton
          name="refreshChannels"
          size="small"
          className="add-icon"
          onClick={getChannels}
        >
          <Refresh />
        </IconButton>
      </div>
    </ServiceForm>
  );
}
const mapStateToProps = (state) => ({
  connections: state.connections,
  channels: state.discord.channels,
});

const mapDispatchToProps = (dispatch) => ({
  getConnections: () => dispatch(startGetConnections()),
  getChannels: () => dispatch(actions.startGetChannels()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Form);
