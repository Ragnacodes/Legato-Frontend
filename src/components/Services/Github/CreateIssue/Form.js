import React, { useState, useLayoutEffect, useEffect } from 'react';
import ServiceForm from '../../../PopoverForm';
import Axios from '../../../../utils/axiosConfig';
import { TextField, MenuItem, IconButton, CircularProgress } from '@material-ui/core';
import { Add } from '@material-ui/icons';
import { startGetConnections } from '../../../../actions/connections';
import { connect } from 'react-redux';

const Form = ({ id, data, editElement, setAnchorEl, getConnections, githubConnections, username }) => {
    
    const [info, setInfo] = useState({
        name: data.name || '',
        connectionId: data.connectionId || '',
        repository: data.repositoryName ? data.owner + '/' + data.repositoryName : '',
        title: data.title || '',
        body: data.body || '',
        labels: data.labels || [],
        assignee: data.assignee || [],
        state: 'open'
    });

    const [repositories, setRepositories] = useState([]);
    const [loadingRepo, setLoadingRepo] = useState(false);
    
    useEffect(() => {
        getConnections();
    }, [getConnections]);

    useLayoutEffect(() => {
        setLoadingRepo(true);
        Axios.post(`/users/${username}/services/github/repositories`, {connectionId: info.connectionId})
            .then((res) => {
                setRepositories(res.data.repositoriesName);
                setLoadingRepo(false);
            })
            .catch((err) => {
                console.log(err);
            });
    }, [info.connectionId, username]);

    const handleChange = (e) => {
        setInfo((prev) => ({
            ...prev,
            [e.target.name]: e.target.value,
        }));
    };
    
    const handleCancel = () => {
        setAnchorEl(null);
    };

    const handleAddConnection = () => {
        window.location.href = "http://localhost:3000/connections";
    }

    const handleSave = () => {
        const updates = {
            name: info.name,
            data: { 
                ...data,
                ...info,
                owner: info.repository.split('/')[0],
                repositoryName: info.repository.split('/')[1],
            }
        };
        editElement(id, updates);
        setAnchorEl(null);
    };

    return (
        <ServiceForm
            className="create-issue"
            title="Create issue"
            disabledSave={false}
            handleSave={handleSave}
            handleCancel={handleCancel}
        >
            <div className="connection-field">
                <TextField
                    name="connectionId"
                    className="text-field"
                    size="small"
                    select
                    label="Connection"
                    value={info.connectionId}
                    onChange={handleChange}
                    variant="outlined"
                    helperText="Choose one of your github connections."

                >
                    { 
                        githubConnections.map((c) => { 
                            return <MenuItem key={c.id} value={c.id}>{c.name}</MenuItem>; 
                        }) 
                    }
                </TextField>
                <IconButton
                    name="addConnection"
                    size="small"
                    className="add-icon"
                    onClick={handleAddConnection}
                >
                    <Add />
                </IconButton>
            </div>
            {
                info.connectionId ? 
                <TextField
                    name="repository"
                    className="text-field"
                    size="small"
                    select
                    label="Repositories"
                    value={info.repository}
                    onChange={handleChange}
                    variant="outlined"
                    helperText="Choose one of your repositories."
                >
                    {
                        loadingRepo ? <CircularProgress color="secondary" />
                        : 
                        repositories.map((r) => { return <MenuItem value={r}>{r}</MenuItem>; }) 
                    }
                </TextField>
                :
                null
            }
            {
                info.repository ?
                <React.Fragment>
                    <TextField
                        name="title"
                        className="text-field"
                        size="small"
                        label="Title"
                        value={info.title}
                        onChange={handleChange}
                        variant="outlined"
                        helperText="Choose the title of the issue."
                    />

                    <TextField
                        name="body"
                        className="text-field"
                        size="small"
                        label="Description"
                        value={info.body}
                        onChange={handleChange}
                        variant="outlined"
                        multiline
                        helperText="Write a description for the issue if you want."
                    />
                </React.Fragment>
                :
                null
            }
        </ServiceForm>
    );
}

const mapStateToProps = (state) => ({
    githubConnections : state.connections.filter((c) => c.type === 'github'),
    username : state.auth.username,
})

const mapDispatchToProps = (dispatch) => ({
    getConnections: () => dispatch(startGetConnections()),
});

export default connect(mapStateToProps, mapDispatchToProps) (Form);