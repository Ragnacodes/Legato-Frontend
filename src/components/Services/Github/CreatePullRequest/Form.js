import React, { useState, useEffect } from 'react';
import ServiceForm from '../../../PopoverForm';
import Axios from '../../../../utils/axiosConfig';
import { TextField, MenuItem, IconButton } from '@material-ui/core';
import { Add } from '@material-ui/icons';
import { startGetConnections } from '../../../../actions/connections';
import { connect } from 'react-redux';

const Form = ({ id, data, editElement, setAnchorEl, getConnections, githubConnections, username }) => {
    
    const [info, setInfo] = useState({
        name: data.name || '',
        connectionId: data.connectionId || '',
        repository: data.repositoryName ? data.owner + '/' + data.repositoryName : '',
        title: data.title || '',
        body: data.body || '',
        base: data.base || '',
        head: data.head || '',
    });

    const [repositories, setRepositories] = useState([]);
    const [branches, setBranches] = useState([]);
    const [loadingRepo, setLoadingRepo] = useState(false);
    const [loadingBranch, setLoadingBranch] = useState(false);
    
    useEffect(() => {
        getConnections();
    }, [getConnections]);

    useEffect(() => {
        setLoadingRepo(true);
        Axios.post(`/users/${username}/services/github/repositories`, {connectionId: info.connectionId})
            .then((res) => {
                setRepositories(res.data.repositoriesName);
                setLoadingRepo(false);
            })
            .catch((err) => {
                console.log(err);
            });
    }, [info.connectionId, username]);

    useEffect(() => {
        setLoadingBranch(true);
        Axios.post(`/users/${username}/services/github/branches`, {
                                                                    connectionId: info.connectionId,
                                                                    repositoryName: info.repository
                                                                  })
            .then((res) => {
                setBranches(res.data.branchesName);
                setLoadingBranch(false);
            })
            .catch((err) => {
                console.log(err);
            });
    }, [info.connectionId, info.repository, username]);

    const handleChange = (e) => {
        setInfo((prev) => ({
            ...prev,
            [e.target.name]: e.target.value,
        }));
    };
    
    const handleCancel = () => {
        setAnchorEl(null);
    };

    const handleAddConnection = () => {
        window.location.href = "http://localhost:3000/connections";
    }

    const handleSave = () => {
        const updates = {
            name: info.name,
            data: { 
                ...data,
                ...info,
                owner: info.repository.split('/')[0],
                repositoryName: info.repository.split('/')[1],
            }
        };
        editElement(id, updates);
        setAnchorEl(null);
    };

    return (
        <ServiceForm
            className="create-pullRequest"
            title="Create pull request"
            disabledSave={false}
            handleSave={handleSave}
            handleCancel={handleCancel}
        >
            <div className="connection-field">
                <TextField
                    name="connectionId"
                    className="text-field"
                    size="small"
                    select
                    label="Connection"
                    value={info.connectionId}
                    onChange={handleChange}
                    variant="outlined"
                    helperText="Choose one of your github connections."

                >
                    { 
                        githubConnections.map((c) => { 
                            return <MenuItem key={c.id} value={c.id}>{c.name}</MenuItem>; 
                        }) 
                    }
                </TextField>
                <IconButton
                    name="addConnection"
                    size="small"
                    className="add-icon"
                    onClick={handleAddConnection}
                >
                    <Add />
                </IconButton>
            </div>
            {
                info.connectionId ?
                <TextField
                    name="repository"
                    className="text-field"
                    size="small"
                    select
                    label="Repositories"
                    value={info.repository}
                    onChange={handleChange}
                    variant="outlined"
                    helperText="Choose one of your repositories."
                >
                    { repositories.map((r) => { return <MenuItem key={r} value={r}>{r}</MenuItem>; }) }
                </TextField>
                :
                null
            }
            {
                info.repository ?
                <React.Fragment>
                    <TextField
                        name="base"
                        className="text-field"
                        size="small"
                        select
                        label="From"
                        value={info.base}
                        onChange={handleChange}
                        variant="outlined"
                        helperText="Choose the base branch."
                    >
                        { branches.map((b) => { return <MenuItem key={b} value={b}>{b}</MenuItem>; }) }
                    </TextField>
                    
                    <TextField
                        name="head"
                        className="text-field"
                        size="small"
                        select
                        label="To"
                        value={info.head}
                        onChange={handleChange}
                        variant="outlined"
                        helperText="Choose the base branch."
                    >
                        { branches.map((b) => { return <MenuItem key={b} value={b}>{b}</MenuItem>; }) }
                    </TextField>
                </React.Fragment>
                :
                null
            }   

        </ServiceForm>
    );
}

const mapStateToProps = (state) => ({
    githubConnections : state.connections.filter((c) => c.type === 'github'),
    username : state.auth.username,
})

const mapDispatchToProps = (dispatch) => ({
    getConnections: () => dispatch(startGetConnections()),
});

export default connect(mapStateToProps, mapDispatchToProps) (Form);