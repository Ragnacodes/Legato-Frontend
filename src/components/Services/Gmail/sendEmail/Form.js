import React, { useState } from 'react';
import ServiceForm from '../../../PopoverForm';
import {
    Grid,
    TextField,
    IconButton,
    Box,
    Button,
    Popover,
    List,
    ListItem,
    ListItemText,
    ListItemSecondaryAction
} from '@material-ui/core';
import { Add, Delete } from '@material-ui/icons';
import AutoSuggestField from '../../../AutoSuggestField';

function validateEmail(email) {
    const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    return re.test(String(email).toLowerCase());
}

const Form = ({ id, data, editElement, setAnchorEl }) => {
    
    const [info, setInfo] = useState({
        name: data.name || '',
        email: data.email || '',
        password: data.password || '',
        to: data.to || [],
        subject: data.subject || '',
        body: data.body || '',
        receiver: '',
    });

    const [listAnchor, setListAnchor] = useState(null);
    const listOpen = Boolean(listAnchor);
    const popoverID = listOpen ? 'simple-popover' : undefined;

    const handleChange = (eventOrValue, targetName) => {
        if (targetName) {
            // AutoSuggestField
            // eventOrValue is value
            // targetName is targetName
            setInfo((prev) => ({
                ...prev,
                [targetName]: eventOrValue,
            }));
        }

        else {
            // MUI TextField
            // eventOrValue is event
            // targetName is undefined
            setInfo((prev) => ({
                ...prev,
                [eventOrValue.target.name]: eventOrValue.target.value,
            }));
        }
    };

    const handleAddEmail = () => {
        setInfo((prev) => {
            return {
                ...prev,
                to: [...prev.to, prev.receiver],
                receiver: ''
            };
        });
    };

    const removeEmail = (recEmail) => {
        const to = info.to.filter(email => email !== recEmail);
        setInfo((prev) => {
            return {
                ...prev,
                to
            };
        });
    };
    
    const handleCancel = () => {
        setAnchorEl(null);
    };

    const handleSave = () => {
        const updates = {
            name: info.name,
            data: { 
                ...data,
                ...info
            }
        };
        editElement(id, updates)
        .then(() => {
            setAnchorEl(null);
        });
    };

    return (
        <ServiceForm
            title="Send Email"
            handleSave={handleSave}
            handleCancel={handleCancel}
            disabledSave={!validateEmail(info.email)}
        >
            <TextField
                error={!validateEmail(info.email)}
                onChange={handleChange}
                label="Email"
                type="email"
                name="email"
                value={info.email}
                variant="outlined"
                size="small"
                margin="normal"
                fullWidth
            />

            <TextField
                onChange={handleChange}
                label="Email Password"
                type="password"
                name="password"
                value={info.password}
                variant="outlined"
                size="small"
                margin="normal"
                fullWidth
            />

            <Grid container alignItems="center">
                <Grid item xs={true}>
                    <Box my={1}>
                        <TextField
                            onChange={handleChange}
                            label="Receiver"
                            type="email"
                            name="receiver"
                            value={info.receiver}
                            variant="outlined"
                            size="small"
                            fullWidth
                        />
                    </Box>
                </Grid>

                <Grid item>
                    <Grid container justify="center">
                        <IconButton
                            color="primary"
                            disabled={!validateEmail(info.receiver)}
                            onClick={handleAddEmail}
                        >
                            <Add />
                        </IconButton>
                    </Grid>
                </Grid>
                <Grid item>
                    <Grid container justify="center">
                        <Button
                            color="primary"
                            onClick={(e) => setListAnchor(e.currentTarget)}
                        >
                            List
                        </Button>
                        <Popover
                            id={popoverID}
                            open={listOpen}
                            anchorEl={listAnchor}
                            onClose={() => setListAnchor(null)}
                            anchorOrigin={{
                                vertical: 'center',
                                horizontal: 'right',
                            }}
                            transformOrigin={{
                                vertical: 'center',
                                horizontal: 'left',
                            }}
                        >
                            <List dense>
                                {
                                    info.to.map((recEmail, index) => {
                                        return (
                                            <ListItem key={index}>
                                                <ListItemText
                                                    primary={recEmail}
                                                />
                                                <ListItemSecondaryAction>
                                                    <IconButton edge="end" onClick={() => removeEmail(recEmail)}>
                                                        <Delete className="error"/>
                                                    </IconButton>
                                                </ListItemSecondaryAction>
                                            </ListItem>
                                        )
                                    })
                                }
                            </List>
                        </Popover>
                    </Grid>
                </Grid>

            </Grid>

            <TextField
                onChange={handleChange}
                label="Subject"
                name="subject"
                value={info.subject}
                variant="outlined"
                size="small"
                margin="normal"
                fullWidth
            />

            <AutoSuggestField
                ancestors={data.ancestors}
                onChange={(value) => handleChange(value, 'body')}
                label="Body"
                name="body"
                value={info.body}
                variant="outlined"
                size="small"
                margin="normal"
                fullWidth
                multiline
            />
        </ServiceForm>
    );
}

export default Form;