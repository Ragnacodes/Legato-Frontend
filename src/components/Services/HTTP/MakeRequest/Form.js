import React, { useState } from 'react';
import ServiceForm from '../../../PopoverForm';
import { TextField, Select, MenuItem } from '@material-ui/core';

const Form = ({ id, data, editElement, setAnchorEl }) => {
    const [info, setInfo] = useState({
        name: data.name || '',
        url: data.url || '',
        method: data.method || 'get',
        body: data.body ? JSON.stringify(data.body) : ''
    });

    const handleChange = (e) => {
        setInfo((prev) => ({
            ...prev,
            [e.target.name]: e.target.value,
        }));
    };
    
    const handleCancel = () => {
        setAnchorEl(null);
    };
    const handleSave = () => {
        if(info.body.length > 0) {
            info.body = JSON.parse(info.body);
        }
        const updates = {
            name: info.name,
            data: { ...data, ...info }
        };
        editElement(id, updates);
        setAnchorEl(null);
    };

    return (
        <ServiceForm
            className="make-request"
            title="Send HTTP request"
            disabledSave={false}
            handleSave={handleSave}
            handleCancel={handleCancel}
        >
            <TextField
                className="text-field"
                name="name"
                label="Name"
                type="text"
                variant="outlined"
                size="medium"
                onChange={handleChange}
                multiline
                value={info.name}
            />

            <TextField
                className="text-field"
                name="url"
                label="URL"
                type="text"
                variant="outlined"
                size="medium"
                onChange={handleChange}
                helperText="write the URL of your HTTP request here."
                multiline
                value={info.url}
            />

            <Select
                className="select"
                name="method"
                variant="outlined"
                size="small"
                onChange={handleChange}
                value = {info.method}
            >
                <MenuItem value="get">GET</MenuItem>
                <MenuItem value="post">POST</MenuItem>
            </Select>

            {
                info.method === "post" ?
                <TextField
                    className="text-field"
                    name="body"
                    label="Body"
                    type="text"
                    variant="outlined"
                    size="small"
                    onChange={handleChange}
                    helperText="write the body of your post request here."
                    multiline
                    value={info.body}
                    placeholder='{"key": "value"}'
                />
                :
                null
            }

        </ServiceForm>
    );
}

export default Form;