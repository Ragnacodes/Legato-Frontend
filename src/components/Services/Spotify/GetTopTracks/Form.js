import React, { useState, useEffect } from 'react';

import ServiceForm from '../../../PopoverForm';
const Form = ({ id, data, editElement, setAnchorEl }) => {
  const [info, setInfo] = useState({
    connection: data.connection || '',
  });

  const [errors, setErrors] = useState({
    connection: !!data.connection,
  });


  useEffect(() => {
    if (!!info['connection']) {
      setErrors((prev) => ({
        ...prev,
        connection: false,
      }));
    }
  }, [info]);


  const handleCancel = () => {
    setAnchorEl(null);
    setInfo({
      connection: data.connection || '',

    });
    setErrors({
      connection: !!data.connection,
    });
  };

  const handleSave = () => {
    editElement(id, { data: { ...data, ...info } });
    setAnchorEl(null);
  };

  return (
    <ServiceForm
      className="watch-playlist"
      title="Watch a Playlist"
      disabledSave={false}
      handleSave={handleSave}
      handleCancel={handleCancel}
    >
      
      
    </ServiceForm>
  );
};

export default Form;
