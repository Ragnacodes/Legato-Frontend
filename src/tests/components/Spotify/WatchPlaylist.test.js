import React from 'react';
import { shallow } from 'enzyme';
import { Form } from '../../../components/Services/Spotify/WatchPlaylist/Form';
import { IconButton, TextField } from '@material-ui/core';

let wrapper,
  props,
  id,
  data,
  connections,
  playlists,
  editElement,
  setAnchorEl,
  getPlaylists;

beforeEach(() => {
  id = 'node_id';
  data = {};
  connections = [];
  playlists = [];
  editElement = jest.fn();
  setAnchorEl = jest.fn();
  getPlaylists = jest.fn();
  props = {
    id,
    data,
    connections,
    playlists,
    editElement,
    setAnchorEl,
    getPlaylists,
  };
  wrapper = shallow(<Form {...props} />);
});

test('should render watch playlist form correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should have 1 textfield for playlist', () => {
  expect(wrapper.find(TextField).find({ name: 'playlist' }).length).toBe(1);
});
