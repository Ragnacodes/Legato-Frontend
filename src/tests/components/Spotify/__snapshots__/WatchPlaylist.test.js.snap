// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render watch playlist form correctly 1`] = `
<PopoverForm
  className="watch-playlist"
  disabledSave={false}
  handleCancel={[Function]}
  handleSave={[Function]}
  title="Watch a Playlist"
>
  <div
    className="playlist-field"
  >
    <WithStyles(ForwardRef(TextField))
      className="text-field"
      label="Playlist"
      name="playlist"
      onChange={[Function]}
      select={true}
      size="small"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(IconButton))
      className="add-icon"
      onClick={[MockFunction]}
      size="small"
    >
      <Memo([object Object]) />
    </WithStyles(ForwardRef(IconButton))>
  </div>
  <WithStyles(ForwardRef(TextField))
    className="text-field"
    error={false}
    helperText="We recommend that you set a low value."
    label="Maximum number of tracks"
    name="max"
    onChange={[Function]}
    size="small"
    type="number"
    value={2}
    variant="outlined"
  />
</PopoverForm>
`;
